x-pack/plugins/fleet/public/applications/integrations/sections/epm/components/package_list_grid/index.tsx

```diff
@@ -0,0 +1,272 @@
+    return {};
+  };
+
+  const splitSubcat = splitSubcategories(availableSubCategories);
+  const { visibleSubCategories } = splitSubcat;
+  const hiddenSubCategoriesItems = useMemo(() => {
+    return splitSubcat?.hiddenSubCategories?.map((subCategory) => {
+      return (
+        <EuiContextMenuItem
+          key={subCategory.id}
+          onClick={() => {
+            onSubCategoryClick(subCategory.id);
+            closePopover();
+          }}
+          icon={selectedSubCategory === subCategory.id ? 'check' : 'empty'}
+        >
+          {subCategory.title}
+        </EuiContextMenuItem>
+      );
+    });
+  }, [onSubCategoryClick, selectedSubCategory, splitSubcat?.hiddenSubCategories]);
+
+  return (
+    <EuiFlexGroup alignItems="flexStart" gutterSize="xl" data-test-subj="epmList.integrationCards">
+      <EuiFlexItem
+        data-test-subj="epmList.controlsSideColumn"
+        grow={1}
+        style={{
+          position: 'sticky',
+          top: 0,
+          zIndex: 100,
+        }}
+      >
+        <ControlsColumn controls={controls} title={title} />
+      </EuiFlexItem>
+      <EuiFlexItem grow={5} data-test-subj="epmList.mainColumn">
+        <SearchBox
+          searchTerm={searchTerm}
+          setSearchTerm={setSearchTerm}
+          selectedCategory={selectedCategory}
+          setCategory={setCategory}
+          categories={categories}
+          availableSubCategories={availableSubCategories}
+          setSelectedSubCategory={setSelectedSubCategory}
+          selectedSubCategory={selectedSubCategory}
+          setUrlandReplaceHistory={setUrlandReplaceHistory}
+        />
+        {showIntegrationsSubcategories && availableSubCategories?.length ? <EuiSpacer /> : null}
+        {showIntegrationsSubcategories ? (
+          <EuiFlexGroup
+            data-test-subj="epmList.subcategoriesRow"
+            justifyContent="flexStart"
+            direction="row"
+            gutterSize="s"
+            style={{
+              maxWidth: 943,
+            }}
+          >
+            {visibleSubCategories?.map((subCategory) => {
+              const isSelected = subCategory.id === selectedSubCategory;
+              return (
+                <EuiFlexItem grow={false} key={subCategory.id}>
+                  <EuiButton
+                    css={isSelected ? 'color: white' : ''}
+                    color={isSelected ? 'accent' : 'text'}
+                    fill={isSelected}
+                    aria-label={subCategory?.title}
+                    onClick={() => onSubCategoryClick(subCategory.id)}
+                  >
+                    <FormattedMessage
+                      id="xpack.fleet.epmList.subcategoriesButton"
+                      defaultMessage="{subcategory}"
+                      values={{
+                        subcategory: subCategory.title,
+                      }}
+                    />
+                  </EuiButton>
+                </EuiFlexItem>
+              );
+            })}
```

Comment:
The actual change for the buttons is in this block, the rest is just splitting the component in smaller one. I extracted some smaller components and moved them to `controls` and `search_box`.